airflow/providers/elasticsearch/log/es_task_handler.py

```diff
@@ -143,7 +143,9 @@ def format_url(host: str) -> str:
         parsed_url = urlparse(host)
 
         # Check if the scheme is either http or https
-        if not parsed_url.scheme:
+        # the second condition handles the Python 3.9+ case where urlparse understands "localhost:9200"
+        # differently than urlparse in Python 3.8 and below (https://github.com/psf/requests/issues/6455)
+        if not parsed_url.scheme or (parsed_url.scheme not in ("http", "https") and not parsed_url.netloc):
```

Comment:
Do you have an example for a case where `parsed_url.scheme` not in ("http", "https") and `parsed_url.netloc` is empty or None?

Reply:
Yes. On Python 3.9 "localhost:9002" parses to :+1: 
* schema="localhost"
* netloc=""

This is why the tests started to fail in main and according to https://github.com/psf/requests/issues/6455 this is expected behaviour :). Don't ask me why - but it is (read the discussion there, it's entertaining).

Reply:
Interestingly we WILL STILL have tests not pass if the url is

* `https:9002`
* `http:9002`

But I figured, that this is a very unlikely edge case.

Reply:
I corrected my questions, I was asking if there is a case where the lib returns `parsed_url.scheme` not in ("http", "https") and `parsed_url.netloc` not empty and not None? and if in this case we should consider the parsed url valid.

I'm asking to know if we should remove `and not parsed_url.netloc`

Reply:
Good question - no idea. But yeah, if `parsed_url.schema is something different that http/https, then you are right - we can safely remove the condition and make it simpler. Let me do that.

Reply:
Fixed. Passes tests. And way simpler to reason about.